src/algorithm/boundingbox.rs

```diff
@@ -0,0 +1,209 @@
+use num::{Float};
+
+use types::{Bbox, Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon};
+
+/// Calculation of the bounding box of a geometry.
+
+pub trait BoundingBox<T: Float> {
+    /// Return a Bounding Box of a geometry
+    ///
+    /// ```
+    /// use geo::{Point, LineString, Coordinate};
+    /// use geo::algorithm::boundingbox::BoundingBox;
+    ///
+    /// let mut vec = Vec::new();
+    /// vec.push(Point::new(40.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 118.34));
+    /// let linestring = LineString(vec);
+    /// let bbox = linestring.bbox().unwrap();
+    ///
+    /// println!("Bbox top left coordinates: {}, {}", bbox.xmin, bbox.ymax);
+    /// println!("Bbox bottom right coordinates: {}, {}", bbox.xmax, bbox.ymin);
+    /// ```
+    ///
+    fn bbox(&self) -> Option<Bbox<T>>;
+}
+
+fn get_bbox<T>(vect: &Vec<Point<T>>) -> Option<Bbox<T>>
+    where T: Float
```

Comment:
On Tue, Jul 19, 2016 at 10:57 AM, Guillaume P. <notifications@github.com>
wrote:
>
> The idea: https://is.gd/iYnUSP
>
> If you have any question, don't hesitate to ask.
>

I've worked on the get_bbox function now it is working accepting an
iterator instead of a vector.
But I'm not able to get the MultiLineString working, if I try with:

get_bbox(&self.0.iter().flat_map(|line| line.0.iter()))

this is the error:

src/algorithm/boundingbox.rs:104:9: 104:17 error: the trait bound
`&std::iter::FlatMap<std::slice::Iter<'_, types::LineString<T>>,
std::slice::Iter<'_, types::Point<T>>, [clo
sure@src/algorithm/boundingbox.rs:104:42: 104:62]>: std::iter::Iterator` is
not satisfied [E0277]
src/algorithm/boundingbox.rs:104
        get_bbox(&self.0.iter().flat_map(|line| line.0.iter()))
                                        ^~~~~~~~
src/algorithm/boundingbox.rs:104:9: 104:17 help: run `rustc --explain
E0277` to see a detailed explanation
src/algorithm/boundingbox.rs:104:9: 104:17 note:
`&std::iter::FlatMap<std::slice::Iter<'_, types::LineString<T>>,
std::slice::Iter<'_, types::Point<T>>, [closure@src/algorith
m/boundingbox.rs:104:42: 104:62]>` is not an iterator; maybe try calling
`.iter()` or a similar method
src/algorithm/boundingbox.rs:104:9: 104:17 note: required because of the
requirements on the impl of `std::iter::IntoIterator` for
`&std::iter::FlatMap<std::slice::Iter<'_, t
ypes::LineString<T>>, std::slice::Iter<'_, types::Point<T>>, [closure@src
/algorithm/boundingbox.rs:104:42: 104:62]>`
src/algorithm/boundingbox.rs:104:9: 104:17 note: required by
`algorithm::boundingbox::get_bbox`
error: aborting due to previous error
Build failed, waiting for other jobs to finish...
src/algorithm/boundingbox.rs:104:9: 104:17 error: the trait bound
`&std::iter::FlatMap<std::slice::Iter<'_, types::LineString<T>>,
std::slice::Iter<'_, types::Point<T>>, [clo
sure@src/algorithm/boundingbox.rs:104:42: 104:62]>: std::iter::Iterator` is
not satisfied [E0277]
src/algorithm/boundingbox.rs:104
        get_bbox(&self.0.iter().flat_map(|line| line.0.iter()))
                                        ^~~~~~~~
src/algorithm/boundingbox.rs:104:9: 104:17 help: run `rustc --explain
E0277` to see a detailed explanation
src/algorithm/boundingbox.rs:104:9: 104:17 note:
`&std::iter::FlatMap<std::slice::Iter<'_, types::LineString<T>>,
std::slice::Iter<'_, types::Point<T>>, [closure@src/algorith
m/boundingbox.rs:104:42: 104:62]>` is not an iterator; maybe try calling
`.iter()` or a similar method
src/algorithm/boundingbox.rs:104:9: 104:17 note: required because of the
requirements on the impl of `std::iter::IntoIterator` for
`&std::iter::FlatMap<std::slice::Iter<'_, t
ypes::LineString<T>>, std::slice::Iter<'_, types::Point<T>>, [closure@src
/algorithm/boundingbox.rs:104:42: 104:62]>`
src/algorithm/boundingbox.rs:104:9: 104:17 note: required by
`algorithm::boundingbox::get_bbox`
error: aborting due to previous error
error: Could not compile `geo`.

I've looked around but I was not able to fix it myself.
So I've commit the last working version. Can you explain me what is wrong
with: get_bbox(&self.0.iter().flat_map(|line| line.0.iter()))?

Thank you for your patience and support!